### PR TITLE
Feature-Update Atomic Habit 

### LINKED ISSUE #23: [User Story Feature] Update atomic habit
**User Story**
As a Computing student with a bad lifestyle, I wish to update the number of times an atomic habit has been carried out.

**Is your feature request related to a problem? Please describe.**
I do not see the progress of my new habit, over time I feel that I have not developed a new habit. Therefore, I wish to be able to track the number of times my atomic habit has been carried out to see my progress and help me to maintain the consistency of my habit.

**Describe the solution you'd like**
Users can use WellNUS++ to update the number of times a habit has been performed.

```update```

WellNUS++ will return this output

```
Here is the current accumulation of your atomic habits!
Keep up the good work and you will develop a helpful habit in no time
1. Make bed every morning - 5
2. Read for at least 30 minutes every day - 3 

Select which habit to update!
```

The user will be prompted to insert the index of habit that they wish to increment

```2```

WellNUS++ will return this output

```
The following habit has been incremented! Keep up the good work!
2. Read for at least 30 minutes every day - 4
```

**Additional context**
Update command will automatically add +1 count to the number of times a habit has been performed

### PR DESCRIPTION
**Pull request by BernardLesley**
Resolves #23 
Add a new feature that allows users to update the count of atomic habits.
Things to do:
1. Add test cases

### CODE REVIEW

File: src/main/java/seedu/duke/atomichabit/command/UpdateCommand.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * Method to execute updating an atomic habit count
+     *
+     * @param atomicHabits
+     * @return CommandResult that contains feedback to the user
+     */
+    @Override
+    public CommandResult execute(AtomicHabitList atomicHabits) throws AtomicHabitException {
+        try {
+            int index = Integer.parseInt(indexString) - 1;
```
Comment (YongbinWang) [inline src/main/java/seedu/duke/atomichabit/command/UpdateCommand.java:30]: Considering changing magic literal 1 to "INDEX_ONE".

File: src/main/java/seedu/duke/atomichabit/command/UpdateCommand.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            AtomicHabit habit = atomicHabits.getHabitByIndex(index);
+            habit.setCount(DEFAULT_INCREMENT);
```
Comment (YongbinWang) [inline src/main/java/seedu/duke/atomichabit/command/UpdateCommand.java:32]: Will user be able to increment by numbers other than 1 in the future?

File: src/test/java/seedu/duke/atomichabit/AtomicHabitTest.java [hunk 3]
```diff
@@ -17,7 +17,7 @@ public class AtomicHabitTest {
      * Test AddCommand with a standard payload to check output printed
      */
     @Test
-    public void addHabit_correct_output() {
+    public void addHabit_correct_output() throws AtomicHabitException{
```
Comment (YongbinWang) [inline src/test/java/seedu/duke/atomichabit/AtomicHabitTest.java:20]: Remember to add basic JUnit testing.

**Review by YongbinWang — COMMENTED** [review]
Code seems to work as intended, will have to integrate with Manager,Command and Ui class. LGTM!

**Review by nichyjt — APPROVED** [review]
Minor formatting nits. LGTM!

File: src/main/java/seedu/duke/atomichabit/command/UpdateCommand.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+public class UpdateCommand extends Command {
+    public static final String COMMAND_WORD = "update";
+    private static final String DOT = ".";
+    private static final int DEFAULT_INCREMENT = 1;
+    private static final String FEEDBACK_STRING = "The following habit has been incremented! Keep up the good work!";
+    private static final String FEEDBACK_INDEX_NOT_INTEGER_ERROR = "Invalid index! Please enter an integer";
+    private static final String FEEDBACK_INDEX_OUT_OF_BOUNDS_ERROR = "Index out of Range! Please enter a valid index";
+    private static final int INDEX_OFFSET = 1;
+    private static final String LINE_SEPARATOR = System.lineSeparator();
```
Comment (haoyangw) [inline src/main/java/seedu/duke/atomichabit/command/UpdateCommand.java:15]: Great work storing magic literals as constants!

File: src/main/java/seedu/duke/atomichabit/feature/AtomicHabit.java [hunk 5]
```diff
@@ -24,6 +24,10 @@ public int getCount() {
         return count;
     }
 
+    public void setCount(int increment){
```
Comment (haoyangw) [inline src/main/java/seedu/duke/atomichabit/feature/AtomicHabit.java:27]: Rmb to add a space before '{' yah!

**Review by haoyangw — APPROVED** [review]
Just one code style error. Other than that everything looks to work as intended, LGTM!

**Pull request merged**